### DISCUSSION TITLE
"dependsOn" functionality on Nova Panel

### DISCUSSION THREAD

**Discussion by syteks** [laravel/nova-issues#4696] [Q&A] [answered]
I am currently using a Panel (\Laravel\Nova\Panel) with a BooleanGroup inside.

The visibility of the BooleanGroup depends on another field and I can hide the BooleanGroup, but I can not hide the Panel, which causes to have an unexpected view.

Because the  Panel is empty and the title is still shown to the user, which can cause confusion.

It would be nice to have a "dependsOn" functionality on the Panel, so we can hide the Panel depending on another Field/Panel.

Current Code and Result:
```             
public function fields(NovaRequest $request)
{

    return [
        Boolean::make("Depending boolean field", 'depending_boolean_field')
            ->hideFromIndex(),
        new Panel('My mystic panel', [
            BooleanGroup::make('Mystical Group', 'attribute')
                ->options([
                    'mystic' => 'Wand',
                    'test' => 'Land',
                ])
                ->dependsOn(['depending_boolean_field'],
                    function (BooleanGroup $field, NovaRequest $request, FormData $formData) {
                        if (! $formData->depending_boolean_field) {
                            $field->hide();
                        }
                    }),
        ]),
    ];
}


```
Result when not checked
![Capture d’écran, le 2022-08-12 à 17 00 17](https://user-images.githubusercontent.com/45171533/184444365-0c6bac9b-96ac-4e1b-ab0d-c8dffcf2b2ae.png)

Expected result when not checked, I would like to have the ability to hide completely the Panel even the Panel's title.

Is there any workaround for the problem?
Or
Could we have something similar to "dependsOn"?

Thank you for your time.

**Comment by crynobone** [comment]
Panel doesn't have any knowledge of field reactive state since it can only access the default `field` value from the initial request.

**Comment by syteks** [comment]
So I imagine there is no easy way to make the panel reactive, is there any chance that we can have a reactive panel with a functionality "belongsTo" or something similar ?

Thank you for your time. 😄

**Reply by crynobone** [reply]
Panel isn't a field, and can't be targeted.

**Reply by syteks** [reply]
Thank you for the fast reply. 😄

**Comment by piljac1** [comment]
@crynobone I have achieved a proof of concept if you're interested (I'll email you the code adjustments if you are). Hopefully we can get it in the core eventually. Here's a gif in action:

![Nova dependant panel](https://i.imgur.com/V5qxuMd.gif)

**Comment by syteks** [comment, marked as answer]
v4.13.0
```Panels will now listen for changes, which allows them to be shown or hidden based on other fields.```

For those wondering, there is no function like dependsOn, but the panels hides automatically when there is no visible field in the panel.

**Reply by sajaddp** [reply]
@S3RGIU-G-T 
Can you give an example of how to use it?
I did not understand how it works from the document.

**Reply by syteks** [reply]
@sajaddp 
In other words all the fields inside the Panel must be hidden in order for the pannel to change state to the hidden state.
So all the fields must have some sort of depends on if we want to dynamically hide the panel.
```
public function fields(NovaRequest $request)
{

    return [
        Boolean::make("Depending boolean field", 'depending_boolean_field')
            ->hideFromIndex(),
            
        new Panel('My mystic panel', [
            BooleanGroup::make('Mystical Group', 'attribute')
                ->options([
                    'mystic' => 'Wand',
                    'test' => 'Land',
                ])
                ->dependsOn(['depending_boolean_field'],
                    function (BooleanGroup $field, NovaRequest $request, FormData $formData) {
                        if (! $formData->depending_boolean_field) {
                            $field->hide();
                        }
                    }),
             Text::make('Mystical Text', 'attribute_text')
                ->dependsOn(['depending_boolean_field'],
                    function (Text $field, NovaRequest $request, FormData $formData) {
                        if (! $formData->depending_boolean_field) {
                            $field->hide();
                        }
                    }),
        ]),
    ];
}
```

**Reply by sajaddp** [reply]
Dear @S3RGIU-G-T, thank you for your help.
